examples/efficientdet/efficientdet_building_blocks.py

```diff
@@ -0,0 +1,321 @@
+        self.data_format = data_format
+        self.resample_layers = []
+        self.vars = []
+
+    def fuse_features(self, nodes):
+        """Fuse features from different resolutions and return a weighted sum.
+
+        # Arguments
+            nodes: a list of tensorflow features at different levels.
+
+        # Returns
+            A tensor denoting the fused features.
+        """
+
+        dtype = nodes[0].dtype
+
+        if self.weight_method == 'attn':
+            edge_weights = [tf.cast(var, dtype=dtype) for var in self.vars]
+            normalized_weights = tf.nn.softmax(tf.stack(edge_weights))
+            nodes = tf.stack(nodes, axis=-1)
+            new_node = tf.reduce_sum(nodes * normalized_weights, -1)
+        elif self.weight_method == 'fastattn':
+            edge_weights = [tf.nn.relu(tf.cast(var, dtype=dtype))
+                            for var in self.vars]
+            weight_sum = add_n(edge_weights)
+            nodes = [nodes[i] * edge_weights[i] / (weight_sum + 0.0001)
+                     for i in range(len(nodes))]
+            new_node = add_n(nodes)
+        else:
+            raise ValueError('unknown weight_method %s' % self.weight_method)
+
+        return new_node
+
+    def _add_wsm(self, initializer, shape=None):
+        for i, _ in enumerate(self.inputs_offsets):
+            self.vars.append(self.add_weight(initializer, shape=shape))
+
+    def build(self, feats_shape):
+        for i, input_offset in enumerate(self.inputs_offsets):
+            name = 'resample_{}_{}_{}'.format(i,
+                                              input_offset,
+                                              len(feats_shape))
+            self.resample_layers.append(ResampleFeatureMap(
+                self.feat_level,
+                self.fpn_num_filters,
+                self.apply_bn_for_resampling,
+                self.conv_after_downsample,
+                self.data_format,
+                name=name
+            ))
+
+        if self.weight_method == 'attn':
+            self._add_wsm('ones')
+        elif self.weight_method == 'fastattn':
+            self._add_wsm('ones')
+        self.op_after_combine = OpAfterCombine(
+            self.conv_bn_act_pattern,
+            self.separable_conv,
+            self.fpn_num_filters,
+            self.act_type,
+            self.data_format,
+            name='op_after_combine{}'.format(len(feats_shape))
+        )
+        self.built = True
+        super().build(feats_shape)
+
+    def call(self, feats, training):
+        nodes = []
+        for i, input_offset in enumerate(self.inputs_offsets):
+            input_node = feats[input_offset]
+            input_node = self.resample_layers[i](input_node, training, feats)
+            nodes.append(input_node)
+        new_node = self.fuse_features(nodes)
+        new_node = self.op_after_combine(new_node)
+        return feats + [new_node]
+
+
+class OpAfterCombine(tf.keras.layers.Layer):
```

Comment:
What is OP standing for. Be explicit about the OP you are programming.

Reply:
It is conv, batchnorm and activation. I named it as ConvolutionAfterCombine